### DISCUSSION TITLE
Difference between the bans and bips tables?

### DISCUSSION THREAD

**Discussion by Ac314** [fail2ban/fail2ban#3174] [General]
Maybe the stupid question: why both the tables bans and bips are present in the database? They seem to contain the same data.

**Comment by sebres** [comment]
Table `bans` contains last banned IPs only, whether table `bips` contains all banned IPs. Both tables will be automatically purged, but from table `bans` it removes tickets older than `dbpurgeage`, whether in table `bips` they remain longer (at least 3 times of the last `bantime` of each ticket). This is made to consider such "bad" IPs in observer module (bantime.increment) in order to ban them faster and for a longer time.
Ultimately a reorganization of database is planned (e. g. separate table for matches etc), so its structure will change soon.

> They seem to contain the same data

Strange, they should not. Is `bantime.increment` off in your configuration?

**Reply by Ac314** [reply]
> Table `bans` contains last banned IPs only, whether table `bips` contains all banned IPs. Both tables will be automatically purged, but from table `bans` it removes tickets older than `dbpurgeage`, whether in table `bips` they remain longer (at least 3 times of the last `bantime` of each ticket). This is made to consider such "bad" IPs in observer module (bantime.increment) in order to ban them faster and for a longer time. Ultimately a reorganization of database is planned (e. g. separate table for matches etc), so its structure will change soon.

Thank you for clarification. I wondered why `Bans` table is purged according to `dbpurgeage` parameter but `Bips` is not. Now it seems everything is OK, I just wait a bit more for `Bips` to be purged.

> 
> > They seem to contain the same data
> 
> Strange, they should not. Is `bantime.increment` off in your configuration?

Sorry, not correctly worded: structure is the same, data format is the same but data itself is different (`Bips` has more records), and I did not understand why 2 similar tables are needed and one of them is not purged in time. Now it is more clear.
